homeassistant/components/sensor/__init__.py

```diff
@@ -565,18 +565,19 @@ def state(self) -> Any:  # noqa: C901
                 )
             return value
 
-        # If the sensor has neither a device class, a state class nor
-        # a unit_of measurement then there are no further checks or conversions
+        prec = self.precision
+
+        # If the sensor has neither a device class, a state class, a unit of measurement
+        # nor a precision then there are no further checks or conversions
         if (
             not device_class
             and not state_class
             and not unit_of_measurement
-            and (prec := self.precision) is None
+            and prec is None
         ):
             return value
 
-        numerical_value: int | float | Decimal | None = None
-        numerical_conversion_failed = None
+        numerical_value: int | float | Decimal
```

Comment:
I think a comment here is useful
```suggestion
        # From here a numerical value is expected
        numerical_value: int | float | Decimal
```